### ISSUE TITLE
Racket language server support

### ISSUE THREAD

**Issue by ghost** [zed-industries/extensions#491]
### Check for existing issues

- [X] Completed

### Language

Racket

### Tree Sitter parser link

https://github.com/6cdh/tree-sitter-racket

### Language server link

https://github.com/jeapostrophe/racket-langserver

### Misc notes

Zed already supports racket syntax highlighting via `tree-sitter-racket` and I wanted to ask if adding the language server would be a welcome contribution before opening a PR (I have a branch and will test it shortly). I also understand if there's a preference to add that language server via a third-party extension instead when that functionality is ready (in which case I'm happy to close the issue!).

**Comment by JosephTLyons** [comment]
Just to note, the racket extension currently resides in the Zed repo and it has tree-sitter support - someone would just need to code to bring in the language server.

**Comment by pseudomata** [comment]
Great, thanks for the note @JosephTLyons! I'd be happy to add code to support the language server in the extension once the migration is complete.

**Comment by pseudomata** [comment]
Thanks for the reference. Would you happen to know if language extensions have a way of setting a "file icon"? e.g. in the previous built-in-to-zed way you could add an `svg` and add that to [assets/icons/file_icons/file_types.json](https://github.com/pseudomata/zed/commit/bef071ff0377e61be0a0ad732d4c9e357a023392#diff-4eb23cf5a8e821ef0f638dc45e40b3d697f0eb17a06e609adb05dd496ca24746): 
```json
"racket": {
    "icon": "icons/file_icons/racket.svg"
}
```
Not critical or anything, just a nice touch 🙂

**Comment by weikenaar** [comment]
Was wondering whether there were any plans for further development.

**Comment by JosephTLyons** [comment]
Closing in favor of:

- https://github.com/zed-industries/extensions/issues/1828

As there was more discussion happening over there.
